cpp/src/arrow/adapters/orc/adapter.h

```diff
@@ -142,6 +143,36 @@ class ARROW_EXPORT ORCFileReader {
   ORCFileReader();
 };
 
+/// \class ORCFileWriter
+/// \brief Write an Arrow Table or RecordBatch to an ORC file.
+class ARROW_EXPORT ORCFileWriter {
+ public:
+  ~ORCFileWriter();
+  /// \brief Creates a new ORC writer.
+  ///
+  /// \param[in] output_stream the io::OutputStream to write into
+  /// \return the returned writer object
+  static Result<std::unique_ptr<ORCFileWriter>> Open(io::OutputStream& output_stream);
```

Comment:
Although the style guide (Google's C++ style guide) has been updated to recommend required output parameter the prevaling style in Arrow's code base is to pass output parameters by pointer.

Reply:
this should clarify ownership semantics (and who needs to call close on output_stream

Reply:
In newer Arrow code we use Result instead of Status & an output parameter.

Reply:
Sure. Just fixed that one.